geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/clustering/mapreduce/SimpleFeatureOutputReducer.java

```diff
@@ -0,0 +1,163 @@
+import mil.nga.giat.geowave.analytics.parameters.ExtractParameters;
+import mil.nga.giat.geowave.analytics.parameters.GlobalParameters;
+import mil.nga.giat.geowave.analytics.tools.AnalyticFeature;
+import mil.nga.giat.geowave.index.StringUtils;
+import mil.nga.giat.geowave.vector.adapter.FeatureDataAdapter;
+
+import org.apache.hadoop.io.ObjectWritable;
+import org.apache.hadoop.mapreduce.ReduceContext;
+import org.apache.hadoop.mapreduce.Reducer;
+import org.opengis.feature.simple.SimpleFeature;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+/**
+ * Remove duplicate input objects and write out as a simple feature with
+ * geometry projected onto CRS EPSG:4326. The output feature contains the ID of
+ * the originating object. The intent is to create a light weight uniform object
+ * that reuses GeoWave data formats to feed analytic processes.
+ * 
+ * If the input object does not require adjustment after de-duplication, use
+ * {@link mil.nga.giat.geowave.accumulo.mapreduce.dedupe.GeoWaveDedupReducer}
+ * 
+ * OutputFeature Attributes, see
+ * {@link mil.nga.giat.geowave.analytics.tools.AnalyticFeature.ClusterFeatureAttribute}
+ *  
+ * Context configuration parameters include:
+ * 
+ * @formatter:off
+ * 
+ * 
+ *                "SimpleFeatureOutputReducer.Extract.DimensionExtractClass" ->
+ *                {@link DimensionExtractor} to extract non-geometric dimensions
+ * 
+ *                "SimpleFeatureOutputReducer.Extract.OutputDataTypeId" -> the
+ *                name of the output SimpleFeature data type
+ * 
+ *                "SimpleFeatureOutputReducer.Global.BatchId" ->the id of the
+ *                batch; defaults to current time in millis (for range
+ *                comparisons)
+ * 
+ * 
+ * @formatter:on
+ */
+
+public class SimpleFeatureOutputReducer extends
+		GeoWaveReducer
+{
+	protected DimensionExtractor<Object> dimExtractor;
+	protected String outputDataTypeID;
+	protected String batchID;
+	protected String groupID;
+	protected FeatureDataAdapter outputAdapter;
+
+	@Override
+	protected void reduceNativeValues(
+			final GeoWaveInputKey key,
+			final Iterable<Object> values,
+			final ReduceContext<GeoWaveInputKey, ObjectWritable, GeoWaveInputKey, Object> context )
+			throws IOException,
+			InterruptedException {
+		final Iterator<Object> valIt = values.iterator();
+		if (valIt.hasNext()) {
+			key.setAdapterId(outputAdapter.getAdapterId());
+			final SimpleFeature feature = getSimpleFeature(
+					key,
+					valIt.next());
+			context.write(
+					key,
+					feature);
+		}
+	}
+
+	private SimpleFeature getSimpleFeature(
+			final GeoWaveInputKey key,
+			final Object entry ) {
+		final Geometry geometry = dimExtractor.getGeometry(entry);
+		final double[] extraDims = dimExtractor.getDimensions(entry);
+
+		final String inputID = StringUtils.stringFromBinary(key.getDataId().getBytes());
+		final SimpleFeature pointFeature = AnalyticFeature.createGeometryFeature(
```

Comment:
is it worth having a feature type that has less attributes, or making all of these unused attributes nillable?
it seems like zoom level, weight, iteration, and count are set with placeholder values?